doc/nrf/ug_thread_commissioning.rst

```diff
@@ -0,0 +1,205 @@
+
+.. toctree::
+   :maxdepth: 1
+
+   ug_thread_commissioning_configuring.rst
+
+.. _thread_ot_commissioning_roles:
+
+Commissioning roles
+*******************
+
+During the commissioning process, the devices involved are assigned one of the following roles:
+
+* Leader - Watches over the network configuration, allows Commissioner Candidate to become Commissioner, and ensures there is only one active Commissioner.
+* Commissioner Candidate - Role for a device capable of becoming the Commissioner, but has not yet been assigned the role of Commissioner.
+* Commissioner - Authenticates the Joiner and provides it with network credentials required to join the network.
+  The Commissioner can be part of the network or it can be outside the network (see :ref:`thread_ot_commissioning_types`).
+* Joiner - Role for a new device that wants to join the Thread network.
+  It exchanges messages with the Commissioner through the directly connected Joiner Router that serves as a proxy.
+* Joiner Router - Role for a router device that is one hop away from the Joiner device in the Thread network and is the sole device connected with the Joiner.
+  Responds to the discovery request of the Joiner.
+  Moreover, when chosen by the Joiner, it passes subsequent communication in a secure manner.
+* Border Router - Role for a device equipped with at least two interfaces (for example Thread and Wi-Fi, Ethernet, LTE etc.) that forwards data between a Thread network and a non-Thread network.
+  It is also an interface for the Commissioner.
+
+  * Border Agent function, usually combined with the Border Router role, accepts :ref:`petitions <thread_ot_commissioning_phases>` from the Commissioner candidate.
+    It also relays commissioning messages between Thread Network and a Commissioner.
+    If a Commissioner is a part of the network, the same node is both the Commissioner and the Border Agent.
+
+  .. note::
+        While the Border Router is typically used only in the external commissioning process, the Nordic Border Router, which is based on the OpenThread Border Router (OTBR), supports both on-mesh and external commissioning.
+
+Once a device is assigned one of the roles in the network, it can combine it with other roles.
+An exception is the Joiner role, which is exclusive to the Joiner and cannot be combined with other roles.
+For example:
+
+* Border Router can also have the Joiner Router role in :ref:`thread_ot_commissioning_types_external`.
+* Joiner Router can also have the Commissioner role in :ref:`thread_ot_commissioning_types_on-mesh`.
+
+For details about scenarios that include devices with multiple roles, see `Thread Group's Commissioning White Paper`_.
+
+See :ref:`thread_ot_commissioning_cli` for the list of commands used to assign the commissioning roles.
+
+.. _thread_ot_commissioning_types:
+
+On-mesh and external commissioning
+**********************************
+
+The commissioning can be either on-mesh or external.
```

Comment:
@edmont, @greg-fer, should we add here that Native Commissioning is not currently supported?

Reply:
I would say yes, we already had this kind of question from users.